src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -267,12 +267,14 @@ private String getTemporaryGcsPath(String bucket, String tableName) {
    */
   private String configureBucket() {
     String bucket = getConfig().getBucket();
-    if (getConfig().getBucket() == null) {
+    if (bucket == null || bucket.trim().isEmpty()) {
       bucket = uuid.toString();
       // By default, this option is false, meaning the job can not delete the bucket.
       // So enable it only when bucket name is not provided.
       baseConfiguration.setBoolean("fs.gs.bucket.delete.enable", true);
     }
+
+    bucket = BigQueryUtil.getBucketNameWithoutScheme(bucket);
```

Comment:
If this is invalid, we should throw an InvalidConfigPropertyException that includes the name of the property that is invalid.